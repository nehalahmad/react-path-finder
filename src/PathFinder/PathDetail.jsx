import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { Alert } from "react-bootstrap";

import { TOTAL_DISTANCE, TOTAL_TIME } from "../config/AppConstants";

/**
 * @description: to display information about distance, and also to show reachable message
 */
const PathDetail = props => {
  const { direction, message } = props;

  return (
    <Fragment>
      {direction && (
        <Fragment>
          {[
            { key: TOTAL_DISTANCE, value: direction.total_distance },
            { key: TOTAL_TIME, value: direction.total_time }
          ].map(({ key, value }, index) => {
            return (
              <Alert variant="success" show={!!value} key={index}>
                <strong>{key}</strong>
                {" " + value}
              </Alert>
            );
          })}
        </Fragment>
      )}
      <Alert variant="danger" show={!!message}>
        {message}
      </Alert>
    </Fragment>
  );
};

// validate prop types
PathDetail.propTypes = {
  message: PropTypes.string,
  direction: PropTypes.object
};

export default PathDetail;
