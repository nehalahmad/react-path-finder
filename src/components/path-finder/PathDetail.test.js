import React from "react";
import { shallow } from "enzyme";

import PathDetail from "./PathDetail";

describe("<PathDetail />", () => {
  it("direction detail is rendered", () => {
    let direction = {
      total_distance: 1000,
      total_time: 500
    };

    const PathDetailComp = shallow(
      <PathDetail direction={direction} />
    );
    expect(PathDetailComp).toMatchSnapshot();
  });

  it("direction detail is rendered", () => {
    const PathDetailComp = shallow(
      <PathDetail message="This location is not accessible by Car" />
    );
    expect(PathDetailComp).toMatchSnapshot();
  });
});
