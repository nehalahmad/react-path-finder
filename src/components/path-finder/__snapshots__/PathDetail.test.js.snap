// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PathDetail /> direction detail is rendered 1`] = `ShallowWrapper {}`;

exports[`<PathDetail /> direction detail is rendered 2`] = `ShallowWrapper {}`;
